chainer/functions/connection/convolution_2d.py

```diff
@@ -183,10 +184,26 @@ def forward_gpu(self, inputs):
                     handle, x_desc.value, filter_desc.value,
                     conv_desc.value, y_desc.value, _fwd_pref, workspace_size)
 
-            if use_tensor_core:
-                # Only CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM
-                # supports Tensor-Core in cuDNN7.
-                algo = libcudnn.CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM  # NOQA
+            _algos = [
+                libcudnn.CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM,
+                libcudnn.CUDNN_CONVOLUTION_FWD_ALGO_WINOGRAD_NONFUSED, ]
+            if use_tensor_core and algo not in _algos:
+                _algo = libcudnn.CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM  # NOQA
+                _size = libcudnn.getConvolutionForwardWorkspaceSize(
```

Comment:
I am not sure how big `getConvolutionForwardWorkspaceSize` is slow, but  how about showing this hint only if `CUDNN_STATUS_BAD_PARAM` error is occurred?

We can create own Exception class to tell `workspace size is not enough`.

Reply:
Thank you for your comment, @sonots.

I understand that your concern is performance degradation due to calling `getConvolutionForwardWorkspaceSize` many times. I believe the execution time of `getConvolutionForwardWorkspaceSize` is not so long, so that it would not give negative impact in performance. In addition, the code path above will be executed only when TensorCore is not used although users want to use TensorCore. Thus, I think it is useful for users to accurately learn the size of necessary cuDNN workspace size for their models and configurations to use TensorCore enabled algorithms in cuDNN.

What would you think on this?

Reply:
> the code path above will be executed only when TensorCore is not used although users want to use TensorCore

The warning is shown only when the case you says, but it looks `getConvolutionForwardWorkspaceSize` is always computed when users want to use TensorCore. 🤔